packages/form-field/src/styles.tsx

```diff
@@ -0,0 +1,84 @@
+import styled, {css} from 'styled-components'
+import {Label as PureLabel, Paragraph, getTextStyle} from '@smashing/typography'
+import {Alert as PureAlert, StyledAlert} from '@smashing/alert'
+import {FormFieldLabelAppearance, FormFieldAlertAppearance} from './types'
+
+export const Description = styled(Paragraph)`
+  margin: 0 0 ${_ => _.theme.spacing.xxxs};
+`
+
+export const Hint = styled(Paragraph)`
+  margin: ${_ => _.theme.spacing.xxxs} 0 0;
+`
+
+interface LabelProps {
+  hasDescription: boolean
+  appearance?: FormFieldLabelAppearance
+}
+
+export const Label = styled(PureLabel)<LabelProps>`
+  ${_ =>
+    _.appearance === 'block' &&
+    css`
+      margin-bottom: ${_.hasDescription ? 0 : _.theme.spacing.xxxs};
+    `}
+
+  ${_ => _.appearance === 'inline' && css``}
+
+  ${_ =>
+    _.appearance === 'overlay' &&
+    css`
+      position: absolute;
+      top: -14px;
+      left: ${_ => _.theme.spacing.xxs};
+      background-color: ${_ => _.theme.colors.background.default};
+      background: ${_ => _.theme.colors.background.default};
+      padding: 0 5px;
+      z-index: 1;
+    `}
+`
+
+interface AlertProps {
+  alertAppearance: FormFieldAlertAppearance
+}
+
+export const Alert = styled(PureAlert)<AlertProps>`
+  margin-top: ${_ => _.theme.spacing.xxs};
+
+  ${_ =>
+    _.alertAppearance === 'overlay' &&
+    css`
+      position: relative;
+      margin-bottom: -18px;
+      top: -18px;
+      margin-left: ${_ => _.theme.spacing.xxs};
+      background-color: ${_ => _.theme.colors.background.default};
+      background: ${_ => _.theme.colors.background.default};
+      padding: 0 5px;
+      z-index: 1;
+      justify-self: flex-start;
+    `}
+
+  ${StyledAlert.Description} {
+    ${getTextStyle({variant: 300, intent: 'danger'})};
+  }
+`
+
+interface FormFieldProps {
+  labelAppearance: FormFieldLabelAppearance
+}
+
+export const FormField = styled.div<FormFieldProps>`
+  position: relative;
+  display: grid;
+  ${_ =>
+    _.labelAppearance === 'inline' &&
+    css`
+      grid-template-columns: 100px 1fr;
```

Comment:
left column is not configurable nor it results from a theme

Reply:
Actually I don't have idea for this. Should it be calculated, passed as a prop or derived from theme or maybe something like this:

```css
grid-template-columns: var(--label-column-width, ${_ => _.labelColumnWidth})
```

It would allow to apply different width for each form section(group of form fields) but fallback to width from props/theme.

```tsx
<Form>
  <section style={{'--label-column-width': 200}}>
    <FormField name="firstName" />
    <FormField name="lastName" />
  </section>
  <section style={{'--label-column-width': 100}}>
    <FormField name="email" />
    <FormField name="password" />
  </section>
</Form>
```

Reply:
I'm a little concerned about variety of possible configuration methods: developer will end up in customizing one part using ThemeProvider and another part using CSS variables. I propose to stick to Theme fields, but make sure that it's configurable by writing custom CSS

Reply:
I've added support for theme, prop and css var configuration